### PR TITLE
.fehbg script now uses absolute file paths instead of the relative path

### LINKED ISSUE #587: .fehbg script can point to wrong location
This line can cause ~/.fehbg to fail when envoking feh from relative paths.
https://github.com/derf/feh/blob/master/src/wallpaper.c#L465

e.g.,
```
> pwd
/home/user
> ./feh/src/feh --bg-center ~/Downloads/image.png
cd ..
> pwd
/home
> ~/.fehbg
/home/avlec/.fehbg: line 2: ./feh/src/feh: No such file or directory
```
~/.fehbg contents
```sh
#!/bin/sh
./feh/src/feh --no-fehbg --bg-center '/home/user/Downloads/image.png'
```

### PR DESCRIPTION
**Pull request by avlec**
Moved the .fehbg script into its own function to improve readability.
Updated so that the script is now generated with an absolute path to the feh executable that was used during the invocation that generate the .fehbg script.

resolves #587

### CODE REVIEW

File: src/wallpaper.c [hunk 1]
```diff
@@ -245,6 +245,96 @@ static void feh_wm_set_bg_maxed(Pixmap pmap, Imlib_Image im, int use_filelist,
...
 }
 
+/*
+** Creates a script that can be used to create the same background
+** as the last time the program was called.
+*/
+void feh_wm_gen_bg_script(char* fil, int centered, int scaled, int filled, int use_filelist) {
+	char * home = getenv("HOME");
+
+	if (home) {
```
Comment (Ferada) [inline src/wallpaper.c:255]: Might as well negate that condition to remove one level of nesting and exit early?

Comment (avlec) [reply src/wallpaper.c:255]: Ingenious, adding change now.

File: src/wallpaper.c [hunk 2]
```diff
@@ -245,6 +245,98 @@ static void feh_wm_set_bg_maxed(Pixmap pmap, Imlib_Image im, int use_filelist,
...
+		return;
+
+	FILE *fp;
+	int fd;
+	char *path;
+	char *exec_absolute_path;
+	char *absolute_path;
+	struct stat s;
+	gib_list *filelist_pos = filelist;
+	exec_absolute_path = feh_absolute_path(cmdargv[0]);
```
Comment (derf) [inline src/wallpaper.c:265]: This only works when running a custom feh build using `./feh`, `src/feh` or similar. `feh_absolute_path` assumes that its argument (if it's not already an absolute path) is relative to the current working directory, so it doesn't work as intended when feh is in $PATH.

In the majority of cases, cmdargv[0] will be just "feh", so .fehbg will contain $pwd/feh – which most likely does not exist.

**Review by derf — CHANGES_REQUESTED** [review]
This sounds useful indeed, but it must only call `feh_absolute_path` when feh was called using an absolute/relative path. When it was run from PATH, cmdargv[0] must be used as-is (as .fehbg will likely have the same PATH as the current shell session).

The distinction should be fairly easy (but please do correct me if I'm missing something):
* if cmdargv[0] contains a '/', feh was called using an absolute/relative path and `feh_absolute_path` is used
* otherwise, feh was called from PATH, and cmdargv[0] is passed as-is

`strchr(cmdargv[0], '/')` should do the trick.

File: src/wallpaper.c (hunk 2 above)
Comment (avlec) [reply src/wallpaper.c]: The code I just pushed should apply the change you suggested. It checks using strchr and then sets the renamed exec_absolute_path, now exec_method, to either the converted path or as raw cmdargv[0]. The memory free also has been updated to ensure cmdargv[0] isn't free'd

Comment (derf) [reply src/wallpaper.c]: Yup, that looks good. Thanks for the patch!

**Review by derf — APPROVED** [review]

**Pull request merged**